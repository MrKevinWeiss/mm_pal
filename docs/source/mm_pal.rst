mm\_pal package
===============

Submodules
----------

mm\_pal.mm\_cmd module
----------------------

.. automodule:: mm_pal.mm_cmd
   :members:
   :undoc-members:
   :show-inheritance:

mm\_pal.mm\_if module
---------------------

.. automodule:: mm_pal.mm_if
   :members:
   :undoc-members:
   :show-inheritance:

mm\_pal.serial\_driver module
-----------------------------

.. automodule:: mm_pal.serial_driver
   :members:
   :undoc-members:
   :show-inheritance:


Module contents
---------------

.. automodule:: mm_pal
   :members:
   :undoc-members:
   :show-inheritance:
