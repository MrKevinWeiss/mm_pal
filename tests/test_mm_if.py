"""Tests Serial Driver for philip pal

    This test should be run with only one PHiLIP device plugged in.
"""
from time import sleep
from pathlib import Path
import pytest
from conftest import MM_PATH, sleep_before_serial_action
from mm_pal import MmIf


def test_parser(mock_app_json, mm_if_inst):
    """Tests the serial driver using the mock device in loopback mode."""
    sleep_before_serial_action()
    mm_if_inst.get_version()
    mm_if_inst.execute_changes()
    mm_if_inst.soft_reset()
    mm_if_inst.read_struct("snum")
    mm_if_inst.read_reg("arr_8")
    mm_if_inst.write_reg("arr_8", [1, 2, 3])


@pytest.mark.parametrize("kwargs", [{},
                                    {'mm_path': MM_PATH},
                                    {'mem_map': []},
                                    {'driver_type': 'serial'},
                                    {"parser_type": 'json'}])
def test_init(mock_app_json, vpr_inst, kwargs):
    mmif = MmIf(port=vpr_inst.ext_port, **kwargs)
    sleep_before_serial_action()
    assert 'version' in mmif.get_version()
    mmif.driver.close()
