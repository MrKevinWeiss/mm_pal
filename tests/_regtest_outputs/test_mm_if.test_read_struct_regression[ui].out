{'ui8': 32}
{'ui16': 8737}
{'ui32': 639968291}
